### PR TITLE
igv: import from homebrew/science

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Companion PR: https://github.com/Homebrew/homebrew-science/pull/6367

File: Formula/igv.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def install
+    inreplace "igv.sh", /^prefix=.*/, "prefix=#{prefix}"
+    prefix.install Dir["igv.sh", "*.jar"]
+    bin.install_symlink prefix/"igv.sh" => "igv"
+  end
+
+  test do
+    ENV.append "_JAVA_OPTIONS", "-Duser.home=#{testpath}/java_cache"
+    (testpath/"script").write "exit"
+    # This command returns 0 on Circle and Travis but 1 on BrewTestBot.
```
Comment (ilovezfs) [inline Formula/igv.rb]: Circle and Travis aren't relevant ;)

Comment (iMichka) [reply Formula/igv.rb]: Woups. Done.

File: Formula/igv.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    inreplace "igv.sh", /^prefix=.*/, "prefix=#{prefix}"
+    prefix.install Dir["igv.sh", "*.jar"]
+    bin.install_symlink prefix/"igv.sh" => "igv"
+  end
+
+  test do
+    ENV.append "_JAVA_OPTIONS", "-Duser.home=#{testpath}/java_cache"
+    (testpath/"script").write "exit"
+    assert_match "Version", `#{bin}/igv -b script`
```
Comment (ilovezfs) [inline Formula/igv.rb]: It should be `shell_output("#{bin}/igv -b script", 1)` if that's the exit code on BrewTestBot. But we need a real test here.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/igv.rb (hunk 2 above)
Comment (iMichka) [reply Formula/igv.rb]: `igv` has a graphical interface, so not sure how we want to test that.

The `-b` flag does the follwing (http://software.broadinstitute.org/software/igv/startingIGV):
> -b <file>, --batch=<file> Immediately run the supplied batch script after start up.

I think that test is not so bad. We could also try to download a file from a server with it and run a batch script on it. Not sure it is worth it.

**Comment by ilovezfs** [comment]
2.4.2 is out

**Comment by iMichka** [comment]
Bumping it in science first: https://github.com/Homebrew/homebrew-science/pull/6385

**Comment by iMichka** [comment]
This is now igv 2.4.2, revision 1.

Should I do something for the test or are you happy with it?

**Comment by iMichka** [comment]
@ilovezfs How do you feel about this? You had still some concerns about the test but I am not sure if I can find anything better, see my comment above.

File: Formula/igv.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/igv.rb]: yes a -b test would be good

File: Formula/igv.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on :java
+
+  def install
+    inreplace "igv.sh", /^prefix=.*/, "prefix=#{prefix}"
+    prefix.install Dir["igv.sh", "*.jar"]
+    bin.install_symlink prefix/"igv.sh" => "igv"
+  end
+
+  test do
+    ENV.append "_JAVA_OPTIONS", "-Duser.home=#{testpath}/java_cache"
```
Comment (ilovezfs) [inline Formula/igv.rb]: not needed

File: Formula/igv.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  revision 1
+  head "https://github.com/broadinstitute/IGV.git"
+
+  bottle :unneeded
+
+  depends_on :java
+
+  def install
+    inreplace "igv.sh", /^prefix=.*/, "prefix=#{prefix}"
+    prefix.install Dir["igv.sh", "*.jar"]
```
Comment (ilovezfs) [inline Formula/igv.rb]: libexec

File: Formula/igv.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+class Igv < Formula
+  desc "Interactive Genomics Viewer"
+  homepage "https://www.broadinstitute.org/software/igv"
+  url "https://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.2.zip"
+  sha256 "d4ba59f9ffd0052d0327ac01fca28b44bb8506ecd504a64783f045c2e7f913a7"
+  revision 1
+  head "https://github.com/broadinstitute/IGV.git"
```
Comment (ilovezfs) [inline Formula/igv.rb]: no

**Comment by ilovezfs** [comment]
:robot:, talk to the hand.

**Comment by iMichka** [comment]
Note to myself: 2.4.4 has been pushed to science. I'll try to find some time today to update this PR.

**Comment by ilovezfs** [comment]
ping

**Comment by iMichka** [comment]
```
igv:
  * C: 14: col 26: Dir(["igv.sh"]) is unnecessary; just use "igv.sh"
```

I'll change that too.

**Comment by iMichka** [comment]
Actually the line is: `libexec.install Dir["igv.sh", "*.jar"]`. Seems the audit does not like that?

**Comment by ilovezfs** [comment]
CC @GauthamGoli hmm that doesn't look right to me.

**Comment by ilovezfs** [comment]
@iMichka the audit will probably be happy with
```
 libexec.install "igv.sh", Dir["*.jar"]
```

**Comment by iMichka** [comment]
Done.

**Comment by ilovezfs** [comment]
```
Error Message

failed: brew test igv --verbose

Stacktrace

        Testing igv
/usr/bin/sandbox-exec -f /tmp/homebrew20171124-98985-ovhzc5.sb /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/igv.rb --verbose
Error: igv: failed
No such file or directory - /usr/local/Cellar/igv/2.4.4_1/libexec/igv
```

**Comment by iMichka** [comment]
All green :)

File: Formula/igv.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+class Igv < Formula
+  desc "Interactive Genomics Viewer"
+  homepage "https://www.broadinstitute.org/software/igv"
+  url "https://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.4.zip"
+  sha256 "a13047b642ad9a09f4a1641e3c99f2785599a3681d93740f67c71d770eb5e62d"
+  revision 1
+
+  bottle :unneeded
+
+  depends_on :java
```
Comment (ilovezfs) [inline Formula/igv.rb]: This should note the minimum version

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by iMichka** [comment]
Done. The minimum is java 8. No clue if this works with java 9.

**Comment by sjackman** [comment]
How's this PR looking?

**Comment by iMichka** [comment]
Good to go for me.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks @iMichka!

**Comment by iMichka** [comment]
Thanks!

**Pull request merged**